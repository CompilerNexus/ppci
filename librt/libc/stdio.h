#ifndef STDIO_H
#define STDIO_H

void printf(char*);
void bsp_putc(char);

#endif

