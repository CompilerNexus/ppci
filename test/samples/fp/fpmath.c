
#include <stdio.h>

float f_pi = 3.14159;
double d_pi = 3.14159;

float f_area(float radius)
{
  return radius * radius * f_pi;
}

double d_area(double radius)
{
  return radius * radius * d_pi;
}

void main_main()
{
  int res;
  res = (int)f_area(3);
  printf("f_area(3) = %d\n", res);
  res = (int)d_area(3);
  printf("d_area(3) = %d\n", res);
}
