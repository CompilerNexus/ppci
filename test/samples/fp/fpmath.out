f_area(3) = 28
d_area(3) = 28
